modules/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/plotly.chart.js

```diff
@@ -0,0 +1,235 @@
+(function ($) {
+
+  const NUMBER_XP_THRESHOLD = 10;
+
+  const PERCENT_DISPLAY_THRESHOLD = 1;
+
+  $.fn.plotlyCodeTypeChart = (title, rootURL, stats, chartDivId) => {
+    var xml = filterStatsByType(stats, 'XML');
+    var java = filterStatsByType(stats, 'JAVA');
+    var javalike = filterStatsByType(stats, 'JAVALIKE');
+    var scriting = filterStatsByType(stats, 'SCRIPTING');
+    var pieData = [xml, java, javalike, scriting];
+    var values = [countStats(xml), countStats(java), countStats(javalike), countStats(scriting)];
+    var total = values.reduce((a, b) => a + b, 0);
+
+    var data = [{
+      values: values,
+      labels: ['XML', 'Java Code', 'Java Like', 'Scripting'],
+      customdata: pieData,
+      type: 'pie'
+    }];
+
+    var div = document.getElementById(chartDivId);
```

Comment:
manipulating this instead of retrieving the element with #getElementById makes the logic misbehave (not sure why) so i kept it as is (still retrieving the id from "this").